pkg/processor/trigger/kafka/types.go

```diff
@@ -286,6 +289,25 @@ func NewConfiguration(id string,
 		}
 	}
 
+	workerTerminationTimeout, err := time.ParseDuration(triggerConfiguration.WorkerTerminationTimeout)
+	if err != nil {
+		return nil, errors.New("Failed to parse default worker termination timeout")
+	}
+
+	if functionconfig.ExplicitAckEnabled(triggerConfiguration.ExplicitAckMode) {
+
+		// pass max worker termination timeout to runtime config
+		if newConfiguration.maxWaitHandlerDuringRebalance > workerTerminationTimeout {
+			runtimeConfiguration.WorkerTerminationTimeout = newConfiguration.maxWaitHandlerDuringRebalance
+		} else {
+			runtimeConfiguration.WorkerTerminationTimeout = workerTerminationTimeout
+		}
```

Comment:
```suggestion
		// pass max worker termination timeout to runtime config
		if newConfiguration.maxWaitHandlerDuringRebalance < workerTerminationTimeout {
			runtimeConfiguration.WorkerTerminationTimeout = newConfiguration.maxWaitHandlerDuringRebalance
		} else {
			runtimeConfiguration.WorkerTerminationTimeout = workerTerminationTimeout
		}
```
Could it be you meant the other way around here? If the `workerTerminationTimeout` is larger than the max, use the max.

Reply:
Nope, `maxWaitHandlerDuringRebalance` is not necessarily longer than `workerTerminationTimeout`, and I want the max between both of them.